azure-mgmt-trafficmanager/src/main/java/com/microsoft/azure/management/trafficmanager/TrafficManagerEndpoint.java

```diff
@@ -225,6 +235,52 @@
         }
 
         /**
+         * The stage of the traffic manager endpoint definition allowing to specify the geographic region.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        @Beta(Beta.SinceVersion.V1_2_0)
+        interface WithGeographicLocation<ParentT> {
+            /**
+             * Specifies the geographic location for the endpoint that will be used when the parent profile is configured
+             * with Geographic based routing method {@link TrafficRoutingMethod#GEOGRAPHIC}.
+             *
+             * @param geographicLocation the geographic location
+             * @return the next stage of the definition
+             */
+            @Beta(Beta.SinceVersion.V1_2_0)
+            WithAttach<ParentT>  withGeographicLocation(GeographicLocation geographicLocation);
+
+            /**
+             * Specifies the list of geographic location for the endpoint that will be used when the parent profile is
+             * configured with Geographic based routing method {@link TrafficRoutingMethod#GEOGRAPHIC}.
+             *
+             * @param geographicLocations the geographic locations
+             * @return the next stage of the definition
+             */
+            @Beta(Beta.SinceVersion.V1_2_0)
+            WithAttach<ParentT> withGeographicLocations(List<GeographicLocation> geographicLocations);
```

Comment:
We have `withGeographicLocations(Collection<String> geographicLocationCodes)` and `withGeographicLocations(List<GeographicLocation> geographicLocations)`, i can't use the Collection for both, as methods have same erasure.